hibernate-core/src/main/java/org/hibernate/type/EntityType.java

```diff
@@ -681,7 +681,7 @@ protected final Object resolveIdentifier(Serializable id, SharedSessionContractI
 				getAssociatedEntityName(),
 				id,
 				eager,
-				isNullable() && !isProxyUnwrapEnabled
+				isNullable()
 		);
 
```

Comment:
Before the change above, ```LoadType``` would end up being ```INTERNAL_LOAD_LAZY``` when enhancement is used with ```@LazyToOne(value = LazyToOneOption.NO_PROXY)```.  See [```SessionImpl#internalLoad```](https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java#L1109-L1116).

```INTERNAL_LOAD_LAZY``` does not allow ```null``. When Hibernate tries to load the entity, it is not found and ```ObjectNotFoundException``` is thrown.

After the change, ```LoadType``` will be ```INTERNAL_LOAD_NULLABLE```, which properly returns ```null```.

Reply:
Sounds reasonable